File: scratch/migrate_products.js
const fs = require('fs');
const path = require('path');

const srcPath = 'c:\\Users\\alienware\\Desktop\\GAMAKAYSHOP\\GAMAKAYSHOP\\src\\lib\\products.ts';
const outputDir = 'c:\\Users\\alienware\\Desktop\\GAMAKAYSHOP\\GAMAKAYSHOP\\src\\data';
const outputPath = path.join(outputDir, 'products.json');

if (!fs.existsSync(outputDir)) {
    fs.mkdirSync(outputDir, { recursive: true });
}

let content = fs.readFileSync(srcPath, 'utf8');

// Extract everything inside the sampleProducts array
const startMarker = 'export const sampleProducts: Product[] = [';
const startIdx = content.indexOf(startMarker) + startMarker.length - 1;
const endMarker = '\n];\n\n// Helper functions';
let endIdx = content.indexOf(endMarker);

if (endIdx === -1) {
    // try alternative end marker
    endIdx = content.lastIndexOf('];');
}

let arrayStr = content.substring(startIdx, endIdx + 1);

// Convert the string representation of a JS array to a real JSON string
// This is still tricky without a full parser, but let's try a better approach:
// Use Function() to evaluate the array literal safely (it's hardcoded data)

try {
    // We need to remove the trailing commas and handle potential TS syntax
    // But evaluating it as a JS expression is usually best for "data in code"
    // We just need to remove comments first
    const cleanArrayStr = arrayStr.replace(/\/\/.*$/gm, '');
    const products = new Function(`return ${cleanArrayStr}`)();
    
    fs.writeFileSync(outputPath, JSON.stringify(products, null, 4), 'utf8');
    console.log(`Successfully created ${outputPath}`);
} catch (e) {
    console.error('Error:', e);
}

File: scratch/migrate_products.py
import json
import re
import os

src_path = r'c:\Users\alienware\Desktop\GAMAKAYSHOP\GAMAKAYSHOP\src\lib\products.ts'
output_dir = r'c:\Users\alienware\Desktop\GAMAKAYSHOP\GAMAKAYSHOP\src\data'
output_path = os.path.join(output_dir, 'products.json')

if not os.path.exists(output_dir):
    os.makedirs(output_dir)

with open(src_path, 'r', encoding='utf-8') as f:
    lines = f.readlines()

products_lines = []
in_array = False
for line in lines:
    if 'export const sampleProducts: Product[] = [' in line:
        in_array = True
        products_lines.append('[')
        continue
    if in_array:
        # Check if the line is exactly ]; (with potential whitespace)
        if line.strip() == '];':
            products_lines.append(']')
            in_array = False
            break
        
        # Clean line: remove trailing comments
        clean_line = line.split('//')[0].strip()
        if not clean_line: continue
        
        # Replace single quotes with double quotes
        clean_line = clean_line.replace("'", '"')
        
        # Add quotes to known keys if they aren't quoted
        keys = ['id', 'name', 'brand', 'category', 'region', 'denominations', 'value', 'price', 'delivery_type', 'image_url', 'logo_url', 'description']
        for key in keys:
            # Use regex to match the key precisely to avoid partial matches
            clean_line = re.sub(fr'\b{key}\b\s*:', f'"{key}":', clean_line)
        
        products_lines.append(clean_line)

json_str = " ".join(products_lines)
# Remove trailing commas inside arrays and objects
json_str = re.sub(r',\s*\]', ']', json_str)
json_str = re.sub(r',\s*\}', '}', json_str)

# One more fix: if there's a comma before the final ], remove it
json_str = json_str.strip()
if json_str.endswith(', ]'):
    json_str = json_str[:-3] + ']'
elif json_str.endswith(',]'):
    json_str = json_str[:-2] + ']'

try:
    data = json.loads(json_str)
    with open(output_path, 'w', encoding='utf-8') as f:
        json.dump(data, f, indent=4, ensure_ascii=False)
    print("Success")
except Exception as e:
    print(f"Error: {e}")
    # Show more context on error
    start = max(0, 32541 - 100)
    end = 32541 + 100
    print(f"Snippet near error: {json_str[start:end]}")
